.github/workflows/ci-test.yml

```diff
@@ -132,6 +166,313 @@ jobs:
+          script: |
+            let pr_data = await github.pulls.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: ${{ needs.prepare.outputs.changes-pull-request-number }}
+            });
+            try {
+              let base_ref = pr_data.data.base.ref;
+              console.log("Base ref:", base_ref);
+              if (!base_ref) {
+                throw "Could not find base ref"
+              }
+              return base_ref
+            } catch (err) {
+              console.log("GH PR payload\n", pr_data);
+              throw err
+            }
+      - name: Setup env
+        run: |
+          BASE_REF=${{steps.pr-base-ref.outputs.result}}
+          echo "BASE_REF=$BASE_REF" >> $GITHUB_ENV
+          echo "BASE_GIT_REV=$(git rev-parse --short=8 origin/$BASE_REF)" >> $GITHUB_ENV
+      - name: find images
+        run: |
+          set +e
+          echo "Will run land_blocking_compat suite"
+          echo "Finding a previous image tag to test against"
+          .github/actions/land-blocking/find-lbt-images.sh > lbt_images_output.txt
+          if [ $? -ne 0 ]; then
+            echo "BUILD_PREV=1" >> $GITHUB_ENV
+            cat lbt_images_output.txt
+            jq -n \
+            --arg msg "*${{ github.job }}* job in ${{ github.workflow }} workflow failed. Could not find a recent image tag for Compat Test" \
+            --arg url "https://github.com/${{ github.repository }}/actions/runs/${{github.run_id}}" \
+            '{
+              "attachments": [
+              {
+                "text": $msg,
+                "actions": [
+                {
+                  "type": "button",
+                  "text": "Visit Job",
+                  "url": $url
+                }
+                ]
+              }
+              ]
+            }' > /tmp/payload
+            curl -X POST -H 'Content-type: application/json' -d @/tmp/payload ${{ secrets.WEBHOOK_FLAKY_LAND_BLOCKING_CT }}
+          else
+            compat_prev_tag=$(tail -1 lbt_images_output.txt)
+            echo "Using previous image tag $compat_prev_tag"
+            echo "PREV_TAG=$compat_prev_tag" >> $GITHUB_ENV
+            echo "BUILD_PREV=0" >> $GITHUB_ENV
+          fi
+      - name: build extra images
+        id: build-extra-images
+        run: |
+          res=land_$BASE_GIT_REV
+          if [ $BUILD_PREV -eq 1 ]; then
+            compat_prev_tag=$res
+            echo "Starting codebuild for $compat_prev_tag"
+            VERSION=$BASE_GIT_REV ADDL_TAG=$compat_prev_tag .github/actions/land-blocking/cti-codebuild.sh &> codebuild-prev.log &
+            prev_build_pid=$!
+            wait $prev_build_pid
+            echo "====== codebuild-prev.log start ======"
+            cat codebuild-prev.log
+          else
+            res=$PREV_TAG;
+          fi
+          echo "::set-output name=prev-tag::$(echo $res)";
+
+  land-blocking-switch:
+    runs-on: ubuntu-latest-xl
```

Comment:
the `xl` size seems wasteful for a trigger condition job.  This can be the default instance size, `ubuntu-latest`